kolibri/core/assets/src/views/k-modal.vue

```diff
@@ -0,0 +1,312 @@
+  import debounce from 'lodash/debounce';
+  import kButton from 'kolibri.coreVue.components.kButton';
+
+  /**
+   * Used to focus attention on a singular action/task
+   */
+  export default {
+    name: 'kModal',
+    components: {
+      kButton,
+    },
+    mixins: [responsiveWindow],
+    $trs: {
+      // error alerts
+      errorAlert: 'Error in',
+      // aria labels
+      closeWindow: 'Close window',
+    },
+    props: {
+      /**
+       * The title of the modal
+       */
+      title: {
+        type: String,
+        required: true,
+      },
+      /**
+       * The text of the submit button
+       */
+      submitText: {
+        type: String,
+        required: false,
+      },
+      /**
+       * The text of the cancel button
+       */
+      cancelText: {
+        type: String,
+        required: false,
+      },
+      /**
+       * Disable the submit button
+       */
+      submitDisabled: {
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * Disable the cancel button
+       */
+      cancelDisabled: {
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * How wide the modal should be.
+       * Small - 300 px.
+       * Medium - 450px.
+       * Large - 100%.
+       */
+      size: {
+        type: String,
+        required: false,
+        default: 'medium',
+        validator(val) {
+          return ['small', 'medium', 'large'].includes(val);
+        },
+      },
+      /**
+       * Toggles error message indicator in title
+       */
+      hasError: {
+        type: Boolean,
+        default: false,
+      },
+    },
+    data() {
+      return {
+        lastFocus: null,
+        maxContentHeight: '1000',
+      };
+    },
+    computed: {
+      modalMaxSize() {
+        return {
+          'max-width': `${this.windowSize.width - 32}px`,
+          'max-height': `${this.windowSize.height - 32}px`,
+        };
+      },
+      contentSectionMaxHeight() {
+        return { 'max-height': `${this.maxContentHeight}px` };
+      },
+    },
+    beforeMount() {
+      this.lastFocus = document.activeElement;
+    },
+    mounted() {
+      this.$nextTick(() => {
+        if (!this.$refs.modal.contains(document.activeElement)) {
+          this.focusModal();
+        }
+      });
+      window.addEventListener('focus', this.focusElementTest, true);
+      window.addEventListener('scroll', this.preventScroll, true);
+    },
+    updated() {
+      this.setContentSectionMaxHeight();
+    },
+    destroyed() {
+      window.removeEventListener('focus', this.focusElementTest, true);
+      window.removeEventListener('scroll', this.preventScroll, true);
+      // Wait for events to finish propagating before changing the focus.
+      // Otherwise the `lastFocus` item receives events such as 'enter'.
+      window.setTimeout(() => this.lastFocus.focus());
+    },
+    methods: {
+      setContentSectionMaxHeight: debounce(function() {
```

Comment:
could you add some comments about the purpose and intended behavior of this function?